proxy_manager.go

```diff
@@ -165,13 +165,19 @@ func (m *proxyManager) removeDistributedObjectEventListener(ctx context.Context,
 	return m.serviceBundle.ListenerBinder.Remove(ctx, subscriptionID)
 }
 
-func (m *proxyManager) remove(serviceName string, objectName string) bool {
+func (m *proxyManager) remove(ctx context.Context, serviceName string, objectName string) bool {
 	name := makeProxyName(serviceName, objectName)
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	if _, ok := m.proxies[name]; !ok {
+	p, ok := m.proxies[name]
+	if !ok {
 		return false
 	}
+	// run the local destroy method of Map
+	if serviceName == ServiceNameMap {
+		mp := p.(*Map)
+		mp.destroyLocally(ctx)
+	}
```

Comment:
Instead of this servicename check we can also do a conditional cast `mp, ok := p.(*Map)` then `if ok { destroyLocally()`